### PR TITLE
D8CORE-2727: js solution for the repeating label issue.

### PR DESCRIPTION
**Pull request by jenbreese**
# READY FOR REVIEW

# Summary
- Removes/hides the duplicate labels when using the load more people.

# Review By (Date)
- 12/3

# Criticality
- high
- SOE, Stanford Basic

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Generate lots of people and sort into a taxonomy term.
3. Verify only the first h3 shows.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [x] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- D8CORE-2727

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: lib/js/stanford_person.behavior.js [hunk 1]
```diff
@@ -21,6 +21,12 @@ export default {
           $(this).attr('aria-expanded', 'false');
         }
       });
+
+      $('.stanford-people-grid--filters h3').each(function(index) {
+        var label = $(this).text();
+        $(this).addClass(index + ':' + label);
```
Comment (pookmish) [inline lib/js/stanford_person.behavior.js:27]: This approach seems fine, but we should change the semicolon and adjust the label before setting the class. Also I don't believe starting with a number is good practice. Perhaps switching them around so the result is `student--index-0`.
Think about the scenario `faculty` and `faculty emeritus` that would result in `<h3 class="0:faculty emeritus">` which would be treated the same as `0:faculty`

We should be able to accomplish that with `label.replace(/[\W_]+/g,"-");`

**Review by pookmish — CHANGES_REQUESTED** [review]

File: lib/js/stanford_person.behavior.js (hunk 1 above)
Comment (jbickar) [reply lib/js/stanford_person.behavior.js:27]: > label.replace(/[\W_]+/g,"-");

Whenever you see a problem and think, "I know, I'll solve this with regular expressions!"

Well, now you have two problems.

Comment (jenbreese) [reply lib/js/stanford_person.behavior.js]: How about this way instead of using classes?

File: lib/js/stanford_person.behavior.js [hunk 2]
```diff
@@ -21,6 +21,25 @@ export default {
...
+        if (headingLabelsCurrent !== prev) {
+          $(this).show();
+        }
+        // to make sure the very first appears.
+        else if (i === 0) {
+          $(this).show();
+        }
+        else {
+          $(this).hide();
+        }
```
Comment (pookmish) [inline lib/js/stanford_person.behavior.js:40]: All this logic can be simplified. Try this suggestion.
```suggestion
        if (headingLabelsCurrent === prev && i > 0) {
          $(this).hide();
        }
```

Comment (jenbreese) [reply lib/js/stanford_person.behavior.js]: Thank you! I updated it and tried on two different taxonomies.

**Review by pookmish — APPROVED** [review]

**Pull request merged**